docs/implementation.md

```diff
@@ -0,0 +1,210 @@
+  new LoggingLink(),
+  new HttpLink(uri);
+])
+```
+
+<p align="center">
+  <br>
+  <img src="images/logging-stack.png" alt="Polling Link"/>
+</p>
+
+To create the `LoggingLink`, we need a way of connecting the next Link.
+The Link's `request` method takes an additional parameter, `forward` that passes an operation to the next Link and returns the resulting Observable.
+`LoggingLink` calls `forward` after printing the operation, then prints the results with the resulting observable.
+
+```js
+class LoggingLink extends ApolloLink {
+
+  request(operation, forward) {
+    console.log(`Operation: ${operation}`);
+
+    const observableResults = forward(operation);
+
+    return observableResults.map((result) => {
+      console.log(`New result: ${result} for ${operation.operationName}`);
+      return result;
+    });
+  }
+}
+```
+
+By using `forward`, `LoggingLink` does not depend on the implementation of the next Link, so can be used with any `ApolloLink`, including `HttpLink`.
+
+**Caution**: when a Link's request function includes `forward`, all the methods of composition expect that the Link will forward the operation.
+
+## Methods of Composition
+
+### Terminating Links
+
+Composing Links distinguishes between Links as terminating or non-terminating.
+A Link is terminating if it does not include `forward` in its signature.
+Each way of connecting Links checks if the current link is non-terminating, then adds the next link.
+Otherwise, the composing function will warn the user.
+This distinction and check is designed to ensure that users who import a Link will not be surprised if an operation is not forwarded.
+
+```js
+request() //terminating
+request(operation) //terminating
+request(operation, forward) //non-terminating
+```
+
+### Single Path - from and concat
+
+`from` is a static method of `ApolloLink` that connects links from an array.
+
+```js
+ApolloLink.from([
+  first,
+  second,
+  third,
+]);
+```
+
+Since all Links extend the abstract class `ApolloLink`, every link contains a `concat` method.
+
+```js
+first.concat(second).concat(third);
+```
+
+`from` and `concat` can be chained in any order:
+
+```js
+ApolloLink.from([first])
+  .concat(second)
+  .concat(ApolloLink.from([third, fourth]))
+  .concat(fifth);
+```
+
+`from` and `concat` will warn the user when attempting to add a Link after a terminating Link.
+
+### Split Points
+
+Links may have may have split points.
+`ApolloLink` contains `split` as a static and instance function.
+A boolean calculated using the operation determines the direction of the split point.
+
+<p align="center">
+  <br>
+  <img src="images/split-link.png" alt="Split Link"/>
+</p>
+
+The default `right` argument is a passthrough link that will call `forward(operation)` and return an empty Observable otherwise.
+A `split` can become a filter by setting `right` to `ApolloLink.empty()`.
+
+```js
+ApolloLink.split(
+  test: (Operation) => boolean,
+  left,
+  right? = ApolloLink.passthrough(),
```

Comment:
Do we actually want people to use this? I feel like it's more confusion than just putting the chain here.